packages/svelte2tsx/src/htmlxtojsx_v2/nodes/Element.ts

```diff
@@ -0,0 +1,253 @@
+import MagicString from 'magic-string';
+import { BaseNode } from '../../interfaces';
+import { surroundWithIgnoreComments } from '../../utils/ignore';
+import { transform, TransformationArray, sanitizePropName } from '../utils/node-utils';
+
+const voidTags = 'area,base,br,col,embed,hr,img,input,link,meta,param,source,track,wbr'.split(',');
+
+/**
+ * Handles HTML elements as well as svelte:options, svelte:head, svelte:window, svelte:body
+ *
+ * Children of this element should call the methods on this class to add themselves to the correct
+ * position within the transformation.
+ *
+ * The transformation result does not have anything to do with HTMLx, it instead uses plan JS,
+ * leveraging scoped blocks (`{ ... }`). Each element is transformed to something that is
+ * contained in such a block. This ensures we can declare variables inside that do not leak
+ * to the outside while preserving TypeScript's control flow.
+ *
+ * A transformation reads for example like this:
+ * ```
+ * // before
+ * <div class={foo} />
+ * // after
+ * { const $$_div = __sveltets_2_createElement("div", {"class": foo,}); }
+ * ```
+ */
+export class Element {
```

Comment:
This is the core of transformations related to HTML elements as well as svelte:options, svelte:head, svelte:window, svelte:body. I hope the comments make it clear how it works.